lua/wikis/commons/MatchPage/Base.lua

```diff
@@ -0,0 +1,263 @@
+end
+
+---@param props {match: MatchGroupUtilMatch}
+---@return Widget
+function BaseMatchPage.getByMatchId(props)
+	local matchPage = BaseMatchPage(props.match)
+	return matchPage:render()
+end
+
+---Tests whether this match page is a Bo1
+---@return boolean
+function BaseMatchPage:isBestOfOne()
+	return #self.matchData.games == 1
+end
+
+---@return Html?
+function BaseMatchPage:getCountdownBlock()
+	if self.matchData.timestamp == DateExt.defaultTimestamp then return end
+	return DisplayHelper.MatchCountdownBlock(self.matchData)
+end
+
+---@private
+---@param site string
+---@param link string
+---@return {icon: string, iconDark: string?, link: string, text: string}?
+function BaseMatchPage._processLink(site, link)
+	return Table.mergeInto({link = link}, Links.getMatchIconData(site))
+end
+
+---Creates an object array for links
+---@private
+---@return {icon: string, iconDark: string?, link: string, text: string}[]
+function BaseMatchPage:_parseLinks()
+	return Array.flatMap(Table.entries(self.matchData.links), function(linkData)
+		local site, link = unpack(linkData)
+		if type(link) == 'table' then
+			return Array.map(link, function(sublink)
+				return BaseMatchPage._processLink(site, sublink)
+			end)
+		end
+		return {BaseMatchPage._processLink(site, link)}
+	end)
+end
+
+---@return (string|Html)[]
+function BaseMatchPage:getVods()
+	local vods =  Array.map(self.games, function(game, gameIdx)
+		return game.vod and VodLink.display{
+			gamenum = gameIdx,
+			vod = game.vod,
+		} or ''
+	end)
+	if String.isNotEmpty(self.matchData.vod) then
+		table.insert(vods, 1, VodLink.display{
+			vod = self.matchData.vod,
+		})
+	end
+	return vods
+end
+
+---@param arr any[]
+---@param item string
+---@return number
+function BaseMatchPage.sumItem(arr, item)
+	return Array.reduce(Array.map(arr, Operator.property(item)), Operator.add, 0)
+end
+
+---@param number number?
+---@return string?
+function BaseMatchPage.abbreviateNumber(number)
+	if not number then
+		return
+	end
+	return string.format('%.1fK', number / 1000)
+end
+
+function BaseMatchPage:populateGames()
+	error('BaseMatchPage:populateGames() cannot be called directly and must be overridden.')
+end
+
+function BaseMatchPage:populateOpponents()
+	Array.forEach(self.opponents, function(opponent, index)
+		opponent.opponentIndex = index
+
+		local teamTemplate = opponent.template and TeamTemplate.getRawOrNil(opponent.template)
+		if not teamTemplate then
+			return
+		end
+
+		opponent.iconDisplay = mw.ext.TeamTemplate.teamicon(opponent.template)
+		opponent.teamTemplateData = teamTemplate
+
+		opponent.seriesDots = Array.map(self.games, function(game)
+			return game.teams[index].scoreDisplay
+		end)
+	end)
+end
+
+function BaseMatchPage:getCharacterIcon(character)
+	error('BaseMatchPage:getCharacterIcon() cannot be called directly and must be overridden.')
+end
+
+function BaseMatchPage:makeDisplayTitle()
+end
```

Comment:
Let's default this imo

Reply:
If you use `noreplace` together with DISPLAYTITLE, a wiki can still override it on the page